### PR TITLE
Convert IBM auth struct fields to pointers

### PR DESCRIPTION
**Pull request by Akurata**
## Problem Statement

The [IBM secret store provider auth](https://github.com/external-secrets/external-secrets/blob/v0.9.4/apis/externalsecrets/v1beta1/secretstore_ibm_types.go#L33) requires either `ContainerAuth` or `SecretRef` to be defined, however they are not struct pointers which when marshaled result in the default `ContainerAuth` value to always be present.

SecretStores that are created programmatically via something like the controller-runtime will always fail to validate because the generated object looks like:
```json
"spec": {
  "controller": "",
  "provider": {
    "ibm": {
      "auth": {
        "secretRef": {
          "secretApiKeySecretRef": {
            "name": "secret-api-key",
            "key": "apikey"
          }
        },
        "containerAuth": {
          "profile": ""
        }
      },
      "serviceUrl": "example.com"
    }
  }
}
```

## Related Issue

None

## Proposed Changes

This PR changes `IBMAuth` field types to be struct pointers. This resolves the marshaling issue which would exclude  omitted `SecretRef` or `ContainerAuth` fields.

## Checklist

- [x] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [x] All commits are signed with `git commit --signoff`
- [x] My changes have reasonable test coverage
- [x] All tests pass with `make test`
- [x] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

**Comment by IdanAdar** [comment]
@moolen Could you review it?

**Review by IdanAdar — APPROVED** [review]

File: pkg/provider/ibm/provider.go [hunk 1]
```diff
@@ -580,28 +582,43 @@ func (ibm *providerIBM) ValidateStore(store esv1beta1.GenericStore) error {
...
+	if missingContainerRef == missingSecretRef {
+		// since both are equal, if one is missing assume both are missing
+		if missingContainerRef {
+			return fmt.Errorf("missing auth method")
 		}
-	} else {
+		return fmt.Errorf("too many auth methods defined")
+	}
+
+	if !missingContainerRef && containerRef.Profile != "" {
```
Comment (moolen) [inline pkg/provider/ibm/provider.go:598]: if profile is empty, we would not get into this branch, hence accessing `secretRef.SecretAPIKey` below which will be a nil pointer dereference panic.

**Review by moolen — CHANGES_REQUESTED** [review]

File: pkg/provider/ibm/provider.go (hunk 1 above)
Comment (Akurata) [reply pkg/provider/ibm/provider.go]: Ah yup, I see what you mean. I've updated the check for the profile to be nested inside the `!missingContainerRef` one and to return an error

**Review by moolen — APPROVED** [review]
LGTM, thank you for the fix!

**Pull request merged**